S1API/Internal/Patches/QuestPatches.cs

```diff
@@ -19,63 +21,80 @@
 using System.Linq;
 using HarmonyLib;
 using Newtonsoft.Json;
-using S1API.Internal.Abstraction;
 using S1API.Internal.Utils;
 using S1API.Quests;
 using UnityEngine;
+using ISaveable = S1API.Internal.Abstraction.ISaveable;
 
 namespace S1API.Internal.Patches
 {
     /// <summary>
-    /// INTERNAL: All patches related to quests.
+    /// INTERNAL: Contains patches specific to quest handling and modification.
     /// </summary>
     [HarmonyPatch]
     internal class QuestPatches
     {
+        protected static readonly Logging.Log Logger = new Logging.Log("QuestPatches");
+
         /// <summary>
-        /// Patching performed when all quests are saved.
+        /// Invoked after all quests are saved.
+        /// Ensures that modded quest data is correctly saved to a designated folder.
         /// </summary>
-        /// <param name="__instance">Instance of the quest manager.</param>
-        /// <param name="parentFolderPath">Path to the base Quest folder.</param>
-        /// <param name="__result">List of extra saveable data. The game uses this for cleanup later.</param>
-        [HarmonyPatch(typeof(S1Quests.QuestManager), "WriteData")]
+        /// <param name="saveFolderPath">The path to the primary save folder where quest data will be stored.</param>
+        [HarmonyPatch(typeof(S1Persistence.SaveManager), nameof(S1Persistence.SaveManager.Save), typeof(string))]
         [HarmonyPostfix]
-        private static void QuestManagerWriteData(S1Quests.QuestManager __instance, string parentFolderPath, ref List<string> __result)
+        private static void SaveManager_Save_Postfix(string saveFolderPath)
         {
-            string questsPath = Path.Combine(parentFolderPath, "Quests");
+            try
+            {
+                string moddedQuestsPath = Path.Combine(saveFolderPath, "..\\Players\\ModdedQuests");
```

Comment:
Do we need to worry about cross-operating systems here with path separators?

Reply:
Agreed, use `Path.Combine` here and don't save in `Players` folder.

Reply:
`Path.Combine` is being used, which is good. I meant use `Path.PathSeparator` too.

Reply:
I originally meant using `Path.PathSeparator` too, but in thinking about it more then I think we just need to use `Path.Combine()` with the `params` overload. Maybe that's what you meant @MaxtorCoder ?